crypto/pkcs11key/key.go

```diff
@@ -30,6 +30,24 @@ var hashPrefixes = map[crypto.Hash][]byte{
 	crypto.RIPEMD160: {0x30, 0x20, 0x30, 0x08, 0x06, 0x06, 0x28, 0xcf, 0x06, 0x03, 0x00, 0x31, 0x04, 0x14},
 }
 
+// Ctx defines the subset of pkcs11.Ctx's methods that we use, so we can inject
+// a different Ctx for testing.
+type Ctx interface {
```

Comment:
Making this a public interface was another example of me forgetting the Go capitalization idiom. I meant to make it private.

However, I think that's orthogonal to your point: by leaving this public and taking it as an arg, we can simplify the code a bit.

One argument against: each place that calls this is going to have to implement the same logic to load and initialize the pkcs11 module. That's a certain amount of duplicated code, but additionally it increases the likelihood that the caller will not bother to keep track of whether they've already initialized the module once in their program lifetime, and introduce bugs accidentally. Implementing it here makes it a small bit easier to use correctly.

On the other hand, if we provide guidance in the docs that the caller should just put the naive, lockless version of this code in their package's init method, maybe it's fine?

Cloudflare folks, do you have a strong opinion either way?